ass2.cpp

```diff
@@ -23,40 +24,53 @@ struct node {
     }
 };
 
-bool have_ancestors(struct node* root, int node_data) {
+bool have_ancestors(struct node* root, int node_data, vector<int> &nodes_to_print) {
     if (root == nullptr) {
         return false;
     }
-    // cout << root -> data << endl;
     
     if (root -> data == node_data) {
         return true;
     }
     
-    if (have_ancestors(root -> left_child, node_data) || have_ancestors(root -> right_child, node_data)) {
-        cout << root -> data << " ";
+    if (have_ancestors(root -> left_child, node_data, nodes_to_print) || have_ancestors(root -> right_child, node_data, nodes_to_print)) {
+        nodes_to_print.push_back(root -> data);
         return true;
     }
     return false;
 }
 
+void assert_output_task1(struct node* root, int node_data, string output, int case_num) {
+    std::stringstream ss;
+    auto old_buf = std::cout.rdbuf(ss.rdbuf());
+    
+    vector<int> nodes_to_print;
+    
+    have_ancestors(root, node_data, nodes_to_print);
+    for(int i = 0; i < nodes_to_print.size(); ++i) {
+        
+        cout << nodes_to_print[i] << " ";
+    }
+    
+    std::cout.rdbuf(old_buf); // reset
+    if (ss.str() == output) {
+        cout << "case " << case_num << " - OK" << endl;
+        return;
+    }
+    cout << "case " << case_num << " - FAIL for the reason: " << endl;
+    if (ss.str() != output) {
```

Comment:
Why this check? If you didn't return inside the previous if this is guaranteed to be true, right?